Makefile

```
@@ -105,11 +111,16 @@ $(BUILD)/test.out: $(BUILD)/test.js | $(BUILD)
 	node $< | tee $@.tmp # Store output in a temp file in case of a failure.
 	mv $@.tmp $@ # Move the output where it belongs.
 
+$(FORMAT)/%.purs.formatted: %.purs $(NODE_MODULES)
+	npx purty --write $<
+	@mkdir -p $$(dirname $@)
+	@touch $@
+
 $(NODE_MODULES): package.json
 	npm install
 	touch $@
 
-$(OUTPUT)/Main/index.js: $(EXAMPLES) $(SRCS) $(BOWER_COMPONENTS) $(NODE_MODULES) | $(BUILD)
+$(OUTPUT)/Main/index.js: format $(EXAMPLES) $(SRCS) $(BOWER_COMPONENTS) $(NODE_MODULES) | $(BUILD)
```

```suggestion
$(OUTPUT)/Main/index.js: $(FORMAT_EXAMPLES) $(FORMAT_SRCS) $(BOWER_COMPONENTS) $(NODE_MODULES) | $(BUILD)
```

Looking at the whole `Makefile`, we're probably better off if we talk about individual files that should exist rather than a phony target. It's more granular, and should be more accurate.